File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.py
class Solution:
    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        n = len(words)
        res = n
        for i in range(n):
            if words[i] == target:
                res = min(res, abs(i - startIndex), i + n - startIndex, startIndex + n - i)
        return -1 if res == n else res

File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.cpp
class Solution {
public:
    int closestTarget(vector<string>& words, string target, int startIndex) {
        int n = words.size();
        int res = n;
        for(int i = 0; i < n; i++)
        {
            if(words[i] == target)
            {
                res = min({res, abs(i - startIndex), i + n - startIndex, startIndex + n - i});
            }
        }
        return res == n ? -1 : res;
    }
};